cpp/src/ray/runtime/task/task_executor.cc

```diff
@@ -150,6 +150,8 @@ Status TaskExecutor::ExecuteTask(
     msgpack::sbuffer sbuf;
 
     if (is_ref_arg) {
+      RegisterOwnershipInfoAndResolveFuture(
+          ref.object_id(), ray::internal::outer_object_id, ref.owner_address());
```

Comment:
`RegisterOwnershipInfoAndResolveFuture` should be invoked every time deserializing an ObjectRef, not just during deserializing task arguments, but also during `ray::Get`. If you invoke it only when deserializing task arguments, how do you plan to support arguments of type `std::vector<ray::ObjectRef>`?